### ISSUE TITLE
Support S3 Acess points

### ISSUE THREAD

**Issue by klizhentas** [gravitational/teleport#4733]
### Description

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

### Environment
- Teleport version (use `teleport version`):
- Tsh version (use `tsh version`):
- OS (e.g. from `/etc/os-release`):

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware):

**Browser environment**

- Browser Version (for UI-related issues):
- Install tools:
- Others:

**Relevant Debug Logs If Applicable**

- `tsh --debug`
- `teleport --debug`

**Comment by zapman449** [comment]
TL;DR: replacing the bucket name with the ARN of the S3 access point triggers a parsing error (it believes the string after the colon to be a port number, since it is parsing as a URL).

Remediation: Support AWS S3 Access Points, document how to use them.

More detail follows

Given the setting:

`audit_sessions_uri: "s3://arn:aws:s3:us-west-2:669997114757:accesspoint/teleport-poc-staging-us-west-2-ap-internal/records"`

where `arn:aws:s3:us-west-2:669997114757:accesspoint/teleport-poc-staging-us-west-2-ap-internal` is the ARN of the S3 access point, `teleport -d` throws a parsing error as follows:

```
ERROR REPORT:
Original Error: *trace.BadParameterError failed to parse URI "s3://arn:aws:s3:us-west-2:669997114757:accesspoint/teleport-poc-staging-us-west-2-ap-internal/records": parse "s3:/arn:aws:s3:us-west-2:669997114757:accesspoint/teleport-poc-staging-us-west-2-ap-internalrecords": invalid port ":accesspoint" after host
Stack Trace:
/go/src/github.com/gravitational/teleport/lib/utils/uri.go:35 github.com/gravitational/teleport/lib/utils.ParseSessionsURI
/go/src/github.com/gravitational/teleport/lib/service/service.go:792 github.com/gravitational/teleport/lib/service.initUploadHandler
/go/src/github.com/gravitational/teleport/lib/service/service.go:972 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initAuthService
/go/src/github.com/gravitational/teleport/lib/service/service.go:651 github.com/gravitational/teleport/lib/service.NewTeleport
/go/src/github.com/gravitational/teleport/lib/service/service.go:394 github.com/gravitational/teleport/lib/service.newTeleportProcess
/go/src/github.com/gravitational/teleport/lib/service/service.go:404 github.com/gravitational/teleport/lib/service.Run
/go/src/github.com/gravitational/teleport/tool/teleport/common/teleport.go:188 github.com/gravitational/teleport/tool/teleport/common.OnStart
/go/src/github.com/gravitational/teleport/tool/teleport/common/teleport.go:165 github.com/gravitational/teleport/tool/teleport/common.Run
/go/src/github.com/gravitational/teleport/tool/teleport/main.go:29 main.main
/opt/go/src/runtime/proc.go:212 runtime.main
/opt/go/src/runtime/asm_amd64.s:1374 runtime.goexit
User Message: failed to parse URI "s3://arn:aws:s3:us-west-2:669997114757:accesspoint/teleport-poc-staging-us-west-2-ap-internal/records": parse "s3://arn:aws:s3:us-west-2:669997114757:accesspoint/teleport-poc-staging-us-west-2-ap-internal/records": invalid port ":accesspoint" after host, initialization failed
```

**Comment by russjones** [comment]
https://github.com/gravitational/teleport/issues/3165

**Comment by zmb3** [comment]
Closing due to inactivity.
